src/main/java/org/elasticsearch/index/query/BaseQueryBuilder.java

```diff
@@ -68,5 +69,14 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
```

Comment:
I think in this case it's not so nice since the existing BaseQueryBuilder already contains some implementations (toString, buildAsBytes etc...) that otherwise we just have to move over. I think to check how many extending classes don't have their own refactored "toQuery()" method it is sufficient to temporarily remove it from the base class and see what turns red. At least that works for me.